pkg/apis/flyteworkflow/v1alpha1/node_status.go

```diff
@@ -128,7 +128,10 @@ func (in *DynamicNodeStatus) SetDynamicNodePhase(phase DynamicNodePhase) {
 }
 
 func (in *DynamicNodeStatus) SetExecutionError(err *core.ExecutionError) {
-	in.Error = &ExecutionError{ExecutionError: err}
+	if err != nil {
+		in.Error = &ExecutionError{ExecutionError: err}
+	}
+	in.Error = nil
```

Comment:
This will reset it all the time... add `else` ?

Reply:
done, good catch